engine/baml-lib/jsonish/src/tests/test_code.rs

```diff
@@ -357,3 +407,220 @@ export default query(async (ctx) => {
+      if (masterExists) return 'master';
+    } catch {
+      throw new Error('Neither main nor master branch found');
+    }
+  }
+
+  throw new Error('Unable to determine main branch');
+}
+
+// Usage
+try {
+  const mainBranch = getMainBranch();
+  console.log(`Main branch is: ${mainBranch}`);
+} catch (error) {
+  console.error(`Error: ${error.message}`);
+}
+"#
+  }
+);
+
+test_deserializer!(
+    triple_quotes_contains_only_backtick_string,
+    BAML_FILE,
+    r#"
+    {
+      "code": ```
+`Hello, world!`
+```,
+      "type": "code",
+    }
+    "#,
+    FieldType::class("Test"),
+    {
+      "type": "code",
+      "code": "`Hello, world!`\n"
+    }
+);
+
+test_deserializer!(
+    triple_backticks_returns_dedented_code_and_discards_info,
+    BAML_FILE,
+    r#"
+Here's a comparison of TypeScript and Ruby code for checking the main Git branch using subprocesses:
+
+{
+  "code": ```typescript main.ts
+    const async function main() {
+      console.log("Hello, world!");
+    }
+```,
+    "type": "code",
+}
+
+  "#,
+  FieldType::class("Test"),
+  {
+    "type": "code",
+    "code": r#"const async function main() {
+  console.log("Hello, world!");
+}
+"#
+  }
+);
+
+// test_deserializer!(
+//     triple_backticks_second_triple_is_not_a_terminator,
+//     BAML_FILE,
+//     r#"
+// Here's a comparison of TypeScript and Ruby code for checking the main Git branch using subprocesses:
+
+// {
+//   "code": ```
+// ``` aaa
+// ```,
+//     "type": "code",
+// }
+
+//   "#,
+//   FieldType::class("Test"),
+//   {
+//     "type": "code",
+//     "code": "``` aaa",
+//   }
+// );
+
+test_deserializer!(
+    triple_backticks_contains_json_terminators,
+    BAML_FILE,
+    r#"
+Here's a comparison of TypeScript and Ruby code for checking the main Git branch using subprocesses:
+
+{
+  "code": ```
+  { type: "code", code: "aaa", closing_terminators: }}}]])) }
+```,
+    "type": "code",
+}
+
+  "#,
+  FieldType::class("Test"),
+  {
+    "type": "code",
+    "code": r#"{ type: "code", code: "aaa", closing_terminators: }}}]])) }
+"#,
+  }
+);
+
+test_deserializer!(
+    triple_backticks_in_json_fenced_codeblock,
+    BAML_FILE,
+    r#"
+Here's a comparison of TypeScript and Ruby code for checking the main Git branch using subprocesses:
+
+```json
+{
+  "code": ```
+  { type: "code", code: "aaa", closing_terminators: }}}]])) }
+```,
+    "type": "code",
+}
+```
+
+  "#,
+  FieldType::class("Test"),
+  {
+    "type": "code",
+    "code": r#"{ type: "code", code: "aaa", closing_terminators: }}}]])) }
+"#,
+  }
+);
+
+test_deserializer!(
+    string_preserves_triple_backticks,
+    BAML_FILE,
+    r#"
+Here's a comparison of TypeScript and Ruby code for checking the main Git branch using subprocesses:
+
+```json
+{
+  "code": "```
+const { execSync } = require('child_process');
+```",
+    "type": "code",
+}
+```
+
+  "#,
+  FieldType::class("Test"),
+  {
+    "type": "code",
+    "code": "```\nconst { execSync } = require('child_process');\n```",
```

Comment:
i thought we said to trim + dedent?

Reply:
This one is triple backticks inside a string, that's why the newlines get preserved!